File: palindrome/palindrome.cpp
/*
Author:  Vaddanak Seng
File:  palindrome.cpp
Purpose:  Determine if user input is a palindrome.
*/

#include <iostream>
#include <string>
//#include "palindrome.h"
#include <sstream>


int main(int argc, char** args) {

	std::stringstream ss;
	size_t result = 0;
	std::string input;
	std::getline(std::cin, input);
	
	const char *start = input.c_str(), *p = start + input.size() - 1;
	
	for( ; p >= start; --p )
		ss <<*p;

	if(input == ss.str())
		result = 1;
	

	//std::cout <<result <<std::endl;
	std::cout <<result;
}

File: palindrome/palindrome.py
#!/usr/bin/env python

u'''
Author: Vaddanak Seng
File: palindrome.py
Purpose: Determine if user input test is a palindrome
'''

from __future__ import print_function;
import sys;


userInput = raw_input(); #get user input

mlist = [x for x in reversed(userInput)]; #convert str into reversed list
userInputReversed = u''.join(mlist);#join list values into a str

if userInput==userInputReversed:
	result = 1;
else:
	result = 0;

print(result, end='');


